src/session.ts

```diff
@@ -28,29 +36,136 @@ export interface Session {
   tx: TxState;
 }
 
+export type ModalState = { status: ModalStateType.Open; modalProps: ModalProps } | { status: ModalStateType.Close; modalProps: null };
+
+export interface ModalProps {
+  config: Config;
+  uri: string;
+}
```

Comment:
These types and vars are specifically used by the connection modal, so they should be named as such, eg. `ConnectionModal`, `ConnectionModalProps` etc.

Reply:
There's still ModalState and ModalStateType that should be renamed